ministryofjustice/correspondence_tool_staff
# CT-4073 Implement SSO with Azure Active Directory

## Description

This PR implements SSO user journey with Azure Active Directory, through the already existing Devise gem and its OmniAuth capability.

The feature can be enabled/disabled through the `enabled_features` mechanism configured in `config/settings.yml`
At the time of this PR it is enabled on Local, for developers to test and play with it.

For more details in Confluence:
https://dsdmoj.atlassian.net/wiki/spaces/STSDD/pages/3925574013/Single+Sign-on

## Self-review checklist




### Screenshots
<img width="760" alt="Screenshot 2022-03-22 at 13 31 09" src="https://user-images.githubusercontent.com/687910/159493016-00f03a23-9e81-4dd1-a058-4d55c3e18e26.png">

<img width="669" alt="Screenshot 2022-03-22 at 13 33 17" src="https://user-images.githubusercontent.com/687910/159493427-f325ff6b-4d49-479c-8845-fc2341d9a42a.png">


### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-4073

### Deployment
This is behind a feature flag and only visible on local environments.

### Manual testing instructions
Once the ENV variables are set, restart the server and the SSO should work with your Microsoft account.
